pkg/block/s3/adapter.go

```diff
@@ -126,7 +156,7 @@ func (a *Adapter) Put(ctx context.Context, obj block.ObjectPointer, sizeBytes in
 		Key:          aws.String(qualifiedKey.Key),
 		StorageClass: opts.StorageClass,
 	}
-	sdkRequest, _ := a.s3.PutObjectRequest(&putObject)
+	sdkRequest, _ := a.s3Client(ctx, qualifiedKey.StorageNamespace).PutObjectRequest(&putObject)
```

Comment:
`qualifiedKey.StorageNamespace` is sometimes different than the bucket.

Reply:
Is it? It used as the bucket for the request 4 lines above